### PR TITLE
chore(ci): new workflow target to handle ci execution from forks

### PR DESCRIPTION
**Pull request by soonum**
closes zama-ai/tfhe-rs-internal#578

### CODE REVIEW

File: .github/workflows/approve_label.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ name: PR approved label manager
 
 on:
   pull_request:
+  pull_request_target:
```
Comment (IceTDrinker) [inline .github/workflows/approve_label.yml:6]: isn't it missing the submitted type ?

**Comment by aquint-zama** [comment]
- add` persist-credentials: false` to checkout actions
- add label `external-contribution`?

otherwise lgtm 👍

**Comment by IceTDrinker** [comment]
> * add` persist-credentials: false` to checkout actions
> * add label `external-contribution`?
> 
> otherwise lgtm 👍

what does the persist credential thing do ?

**Comment by aquint-zama** [comment]
> > * add` persist-credentials: false` to checkout actions
> > * add label `external-contribution`?
> > 
> > otherwise lgtm 👍
> 
> what does the persist credential thing do ?

it prevents adding the token to the git/config (see [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/))

**Comment by IceTDrinker** [comment]
> it prevents adding the token to the git/config (see [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/))

how is it not the default if they know it's completely broken O_o

**Comment by aquint-zama** [comment]
> > it prevents adding the token to the git/config (see [here](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/))
> 
> how is it not the default if they know it's completely broken O_o

good question

**Comment by IceTDrinker** [comment]
I'm not a pro of the security stuff, so looks ok from afar, I let @aquint-zama give the final go ahead as for the credentials stuff, I guess all checkout instances could have the no persist for credentials but it could added in a further PR

**Review by aquint-zama — APPROVED** [review]
lgtm

**Pull request merged**